api/src/websocket/handlers/subscribe.ts

```diff
@@ -162,8 +162,12 @@ export class SubscribeHandler {
 					subscription.event = message.event as SubscriptionEvent;
 				}
 
-				if ('query' in message) {
-					subscription.query = sanitizeQuery(message.query!, accountability);
+				if (message.query) {
+					if (typeof message.query === 'object' && !Array.isArray(message.query)) {
```

Comment:
If im reading this right the WebSocket error gets thrown when the `query` is an object and not is an array. This should be the other way around:
```suggestion
					if (typeof message.query !== 'object' || Array.isArray(message.query)) {
```

Reply:
That is correct 😅 , looks like i got them mixed up with the passing condition